framework/src/meshgenerators/CombinerGenerator.C

```diff
@@ -246,8 +246,10 @@ CombinerGenerator::copyIntoMesh(UnstructuredMesh & destination, const Unstructur
     // Note: if performance becomes an issue, this is overkill for just getting the max node id
     std::set<subdomain_id_type> source_ids;
     std::set<subdomain_id_type> dest_ids;
-    source.subdomain_ids(source_ids, false);
-    destination.subdomain_ids(dest_ids, false);
+    source.subdomain_ids(source_ids, true);
+    destination.subdomain_ids(dest_ids, true);
+    mooseAssert(source_ids.size(), "Should have a subdomain");
+    mooseAssert(dest_ids.size(), "Should have a subdomain");
```

Comment:
Why was this change needed?

Reply:
that's a fix for distributed meshes which sometimes has subdomains not defined on one of the procs